services/wallet/router_v2.go

```diff
@@ -0,0 +1,572 @@
+package wallet
```

Comment:
is the logic of this file mostly a copy/paste from v1 adapted?

Reply:
What I did, I literally copied `router.go` file and then went through the router logic and reorganized some things taking care to not miss anything that we already have there. Guess this way is also easier to compare with the current router.